### PR TITLE
docs: add Ubuntu 16.04 pip installation notes

### PR DESCRIPTION
**Pull request by nvgoldin**
1. Ubuntu 16.04 installation instructions:

With regard to debian vs ubuntu: looks like debian 8 will not work as it has an old version of libguestfs (v1.28), so debian 9 will probably work, but I don't have a VM to test it on at the moment.

If anyone wants to give it a try:

Create ``init-ubu.yaml``:
```yaml
domains:

  ubu:
    memory: 2048
    nics:
      - net: net01
    disks:
      - template_name: ubuntu16.04-base
        type: template
        name: root
        dev: sda
        format: qcow2
      - name: data
        type: empty
        format: qcow2
        size: 50G
        dev: sdb
    metadata:
      deploy-scripts:
        - $LAGO_INITFILE_PATH/install_lago_debian.sh
    artifacts:
      - /var/log
nets:
  net01:
    type: nat
    dhcp:
      start: 100
      end: 254
    management: true
    dns_domain_name: lago.local
```

Create ``install_lago_debian.sh`` in the same dir:
```
#!/bin/bash -ex
readonly DUMMY_USER="koko"

apt-get update
apt-get install -y python python-dev python-pip build-essential libssl-dev \
    libffi-dev qemu-kvm libvirt-bin libvirt-dev pkg-config \
    libguestfs-tools libguestfs-dev git

mkfs.ext4 /dev/sdb
mkdir -p /var/lib/lago
mount /dev/sdb /var/lib/lago
mkdir -p /var/lib/lago/{store,repos,subnets}

adduser --disabled-password --gecos "" "$DUMMY_USER"
chown -R "$DUMMY_USER": /var/lib/lago

usermod -a -G libvirtd,kvm "$DUMMY_USER"
chmod g+x "/home/$DUMMY_USER"
#systemctl enable libvirtd
systemctl restart libvirtd
chmod 0644 /boot/vmlinuz*

pip install -U pip
pip install http://download.libguestfs.org/python/guestfs-1.36.4.tar.gz
pip install --extra-index-url https://testpypi.python.org/pypi lago
```
Run:
```bash
lago init init-ubu.yaml
lago start
lago deploy
```
Log into the ubuntu VM:
```bash
lago shell ubu
```

Then inside the VM:
```bash
su -l koko
cat >> init-nested.yaml << EOF
domains:
  nested:
    memory: 512
    service_provider: systemd
    nics:
      - net: lago-nested
    disks:
      - template_name: el7.3-base
        type: template
        name: root
        dev: vda
        format: qcow2
nets:
  lago-nested:
    gw: 192.168.230.0
    type: nat
    dhcp:
      start: 100
      end: 254
    management: true
    dns_domain_name: lago-nested.local
EOF
```
And finally(still inside the ubu VM):
```
lago init init-nested.yaml
lago start
```

2. Added `python-devel` package to all distros

fixes: https://github.com/lago-project/lago/issues/625
Signed-off-by: Nadav Goldin <ngoldin@redhat.com>

### CODE REVIEW

**Comment by nvgoldin** [comment]
Rendered docs:
http://jenkins.ovirt.org/job/lago_master_github_check-patch-el7-x86_64/478/artifact/exported-artifacts/docs/html/Installation.html

**Comment by eedri** [comment]
We should ask on ovirt devel if anyone can test this ( probably on lago-devel as well )

**Comment by nvgoldin** [comment]
Sure, I already sent an email to lago-devel. I'll send another one after the release(or with the release notes).

**Comment by nvgoldin** [comment]
@gbenhaim - you ok with merging it? Think it would be nice to have the updated docs with the v0.40 release.

**Comment by gbenhaim** [comment]
Too late :-\

**Comment by nvgoldin** [comment]
When you go to lago.readthedocs.org you go by default to 'latest' so it doesn't matter with regard to the tag.

**Comment by nvgoldin** [comment]
So more to the point, Need a review here and we can merge :)

**Comment by gbenhaim** [comment]
1. What's not working in libguestfs on debain 8, sysprep ? 
2. Was `Apparmor` running in the VM when you verified the installation ? I remember that I've add to configure it in order to store images in `/var/lib/lago/store`

```
- Apparmor configuration: if you want to store lago images under /var/lib/lago/store, you will need to configure apparmor:

 apt-get install apparmor-utils; aa-complain /usr/lib/libvirt/virt-aa-helper
```

File: docs/Installation.rst [hunk 1]
```diff
@@ -35,8 +31,14 @@ pip
        $ dnf install -y libvirt libvirt-devel libguestfs-tools \
            libguestfs-devel gcc libffi-devel openssl-devel qemu-kvm
 
-   C. Debian / Ubuntu - *TO-DO*
-   D. ArchLinux - *TO-DO*
+   C. Ubuntu 16.04+
```
Comment (gbenhaim) [inline docs/Installation.rst]: Whet about Debian?

File: docs/Installation.rst [hunk 2]
```diff
@@ -54,11 +56,21 @@ pip
...
-       $ sudo usermod -a -G USERNAME qemu
-       $ chmod g+x $HOME
+     .. code:: bash
+
+         $ sudo usermod -a -G qemu,libvirt USERNAME
+         $ sudo usermod -a -G USERNAME qemu
+         $ sudo chmod g+x $HOME
+
+
+   * Debian :
```
Comment (gbenhaim) [inline docs/Installation.rst]: I think that Ubuntu also needs this config.

**Comment by gbenhaim** [comment]
https://github.com/lago-project/lago/issues/626

File: docs/Installation.rst (hunk 1 above)
Comment (nvgoldin) [inline docs/Installation.rst]: Actually should have been "Debian and its derivatives" but as I only tested this on Ubuntu 16.04, I changed it to Ubuntu 16.04 now.
When we test on Debian 9 - we can change it.

**Comment by nvgoldin** [comment]
> What's not working in libguestfs on debain 8, sysprep ?

Yep. It has libguestfs 1.28 which does not support --commands-from-file. Debian 9 has a newer version and is also available on virt-builder, so we just need to build it and test.

> Was Apparmor running in the VM when you verified the installation ? I remember that I've add to configure it in order to store images in /var/lib/lago/store

Nope, but I think this is similar to 'polkit', no? If you install Lago on Fedora desktop, it will open a pop-up window that you need to confirm the access. I guess later we can add Polkitd/AppAramor instructions as well.

Also added 'python-devel' packages now, to fix https://github.com/lago-project/lago/issues/625

File: docs/Installation.rst [hunk 3]
```diff
@@ -234,3 +246,4 @@ Troubleshooting
        Debian `python-guestfs`.
```
Comment (gbenhaim) [inline docs/Installation.rst]: Since you removed all the docs about Debian,  s/Debian/Ubuntu

Comment (nvgoldin) [inline docs/Installation.rst]: Right, done.

**Comment by gbenhaim** [comment]
ci merge please

**Pull request merged**